python/cuml/ensemble/randomforest.pyx

```diff
@@ -168,13 +179,17 @@ cdef class RandomForest_impl():
         if self.rf_classifier64 != NULL:
             del self.rf_classifier64
 
-        y_m, y_ptr, _, _, y_dtype = input_to_dev_array(y)
+        y_m, y_ptr, _, _, y_dtype = \
+            input_to_dev_array(y, check_dtype=np.int32,
+                               convert_to_dtype=(self.dtype if convert_dtype
+                                                 else None))
 
         if y_dtype != np.int32:
             raise TypeError("The labels need to have dtype = np.int32")
 
         X_m, X_ptr, n_rows, self.n_cols, self.dtype = \
-            input_to_dev_array(X, order='F')
+            input_to_dev_array(X, order='F', check_dtype=[np.float32,
```

Comment:
I get where this is coming from, but it's kind of weird that convert_to_dtype only applies to `y` and not X?

Reply:
Decision was for memory usage, since `X` will be bigger always if we convert it instead it it'll hurt us in performance more than if we convert `y`

Reply:
Oops, the self.dtype is derived from X, so actually the error her is having X converted after y! Will fix that

Reply:
Ah nevermind, the reason here is that the labels need to be `int32` while the `X` can be float or double, so I'm a better programmer initially than at reviewing responses :P